homeworks/zsoltcsengeri/2_vars_datatypes/user_info.py

```diff
@@ -0,0 +1,81 @@
+from pprint import pprint
+
+# The dictionary
+user_info = {
+    "name": "Mike",
+    "age": 25,
+    "favourite_meals": ["pizza", "carbonara", "sushi"],
+    "phone_contacts": {
+        "Mary": "+36701234567",
+        "Tim": "+36207654321",
+        "Tim2": "+36304567321",
+        "Jim": "+364005000",
+    },
+}
+
+
+# 1 Request 4 programming languages from the user (comma-separated)
+programming_languages = input(
+    "Please give me 4 programming languages comma-separated: "
+)
+
+print(type(programming_languages))
+
+# Convert input string to a list. Another method to convert to list is to use list() function:list(programming_languages)
+programming_languages_list = programming_languages.split(",")
+print(programming_languages_list)
+print(type(programming_languages_list))
+
+# Add 'skills' key to user_info dictionary
+user_info["skills"] = programming_languages_list
+pprint(user_info)
+
+# 2 Sort the favourite_meals list alphabetically
+user_info["favourite_meals"].sort()
+pprint(user_info)
+
+# 3 Print the sorted list and the second-to-last element
+print(user_info["favourite_meals"][-2])
+
+# 4 Add 'spaghetti' to favourite_meals
+user_info["favourite_meals"].append("spaghetti")
+print(user_info["favourite_meals"])
+pprint(user_info)
+
+# 5 Add the 3rd and 4th elements again to favourite_meals (based on current list)
+user_info["favourite_meals"].extend(user_info["favourite_meals"][2:4])
+print(user_info["favourite_meals"])
+
+# 6 Remove duplicates (note: set() removes duplicates but changes order)
+user_info["favourite_meals"] = list(set(user_info["favourite_meals"]))
+"""user_info["favourite_meals"] = list(dict.fromkeys(user_info["favourite_meals"]))
```

Comment:
ez is nagyon szuper, hogy tudod hogy a set megváltoztathatja a sorrendet, ezért utánanéztél a dict.fromkeys()-nek!

Reply:
Szia István!

Köszönöm a visszajelzést! Kijavítottam a ternary-s logikát és hozzáadtam a többi javaslatot is. Kérlek, nézd át, hogy minden rendben van. Köszönöm szépen!

Üdv,
Zsolt